dymos/load_case.py

```diff
@@ -51,83 +78,101 @@ def load_case(problem, previous_solution):
+            prev_control_val = prev_vars[prev_control_path]['value']
+            prev_control_units = prev_vars[prev_control_path]['units']
             problem.set_val(control_path,
-                            phase.interpolate(xs=prev_time, ys=prev_control_val,
+                            phase.interpolate(xs=prev_time_val, ys=prev_control_val,
                                               nodes='control_input', kind='slinear'),
                             units=prev_control_units)
 
         # Set the output polynomial control outputs from the previous solution as the value
-        for polynomial_control_name, options in phase.polynomial_control_options.items():
-            polynomial_control_path = [s for s in phase_outputs if
-                                       s.endswith(f'{phase_name}.polynomial_controls:{polynomial_control_name}')][0]
-            prev_polynomial_control_path = [s for s in prev_outputs if
-                                            s.endswith(f'{phase_name}.'
-                                                       f'polynomial_controls:{polynomial_control_name}')][0]
-            prev_polynomial_control_val = prev_outputs[prev_polynomial_control_path]['value']
-            prev_polynomial_control_units = prev_outputs[prev_polynomial_control_path]['units']
+        for pc_name, options in phase.polynomial_control_options.items():
+            polynomial_control_path = [s for s in phase_vars if
+                                       s.endswith(f'{phase_name}.polynomial_controls:{pc_name}')][0]
+            prev_polynomial_control_path = [s for s in prev_vars if
+                                            s.endswith(f'{phase_name}.polynomial_controls:{pc_name}')][0]
+            prev_polynomial_control_val = prev_vars[prev_polynomial_control_path]['value']
+            prev_polynomial_control_units = prev_vars[prev_polynomial_control_path]['units']
             problem.set_val(polynomial_control_path, prev_polynomial_control_val,
                             units=prev_polynomial_control_units)
 
         # Set the timeseries parameter outputs from the previous solution as the parameter value
-        for parameter_name, options in phase.parameter_options.items():
-            parameter_path = [s for s in phase_inputs if s.endswith(f'{phase_name}.parameters:{parameter_name}')][0]
-            prev_parameter_path = [s for s in prev_outputs if
-                                   s.endswith(f'{phase_name}.timeseries.parameters:{parameter_name}')][0]
-            prev_parameter_val = prev_outputs[prev_parameter_path]['value'][0, ...]
-            prev_parameter_units = prev_outputs[prev_parameter_path]['units']
-
-            problem.set_val(parameter_path, prev_parameter_val, units=prev_parameter_units)
+        for param_name, options in phase.parameter_options.items():
+            prev_match = [s for s in prev_vars if s.endswith(f'{phase_name}.parameters:{param_name}')]
```

Comment:
I guess this works without filtering out the traj_params, since that option isn't used.

Reply:
traj params are handled outside of the phase loop, but work by adding a parameter to each phase to which that param is passed.  This catches both, but ultimately as long as the value of those parameters is set at the phase level the value should get set in the phase via promotion.